File: leetcode/938.cpp
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};

class Solution {
private:
  int sum = 0;

public:
  int rangeSumBST(TreeNode *root, int low, int high) {
    if (root == nullptr) {
      return 0;
    }
    if ((root->val >= low) && (root->val <= high)) {
      sum += root->val;
    }
    rangeSumBST(root->left, low, high);
    rangeSumBST(root->right, low, high);
    return sum;
  }

  int rangeSumBST_(TreeNode *root, int low, int high, int sum = 0) {
    if (root == nullptr) {
      return sum;
    }
    if ((root->val >= low) && (root->val <= high)) {
      sum += root->val;
    }
    sum = rangeSumBST_(root->left, low, high, sum);
    sum = rangeSumBST_(root->right, low, high, sum);
    return sum;
  }
};

File: leetcode/938.py
# Definition for a binary tree node.
class TreeNode:

    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:

    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        if not root:
            return 0
        rst = 0
        if root.val >= low and root.val <= high:
            rst += root.val
        if root.val >= low:
            rst += self.rangeSumBST(root.left, low, high)
        if root.val <= high:
            rst += self.rangeSumBST(root.right, low, high)
        return rst